src/app/actions.js

```diff
@@ -139,6 +160,19 @@ export function logOut(getState) {
 */
 export function connectAndStartPolling(reduxStore, bakery) {
   return async function connectAndStartPolling(dispatch) {
-    connectAndListModels(reduxStore, bakery);
+    let additionalControllers = null;
+    try {
+      const data = window.localStorage.getItem("additionalControllers");
+      if (data) {
+        additionalControllers = JSON.parse(data);
+        additionalControllers.forEach((controller) => {
+          dispatch(storeUserPass(controller[0], controller[1]));
+        });
+      }
+    } catch (e) {
+      // XXX Add to Sentry.
+      console.log("oops bad data in the additionalControllers localStorage");
```

Comment:
Might not just be bad data, maybe something like "Error retrieving additional controllers". Then is it worth clearing the localstorage to regain a working UI without the user having the clear the storage themselves?

Reply:
This is an interesting problem...If there is bad data they would have to go in and manually edit/clear it out which isn't a great UX. However I don't want to just clear it as then all evidence of what caused the issue would be erased. 

I'll update the error message here and add another todo around surfacing this to the user somehow so that they can take appropriate action.